index.d.ts

```diff
@@ -907,7 +907,7 @@ type AppendKeyPrefix<Key, KPrefix> = KPrefix extends string
  * T function declaration *
  **************************/
 export interface TFunction<Ns extends Namespace = _DefaultNamespace, KPrefix = undefined> {
-  $TFunctionBrand: Ns;
+  $TFunctionBrand: [Ns] extends [string] ? `${Ns}` : never;
```

Comment:
why not `$TFunctionBrand: [Ns] extends [string] ? `${Ns}` : $FirstNamespace<Ns>;` ?

Reply:
Can you explain? Why FirstNamespace?

Reply:
BTW you can use double-\`... like ``` `` ...`${Ns}`... `` ``` :)

Reply:
https://github.com/i18next/react-i18next/blob/0a4681e428c888fe986bcc0109eb19eab6ff2eb3/src/useTranslation.js#L62

Reply:
Looks like that depends on `i18nOptions.nsMode`, which we can't really check at compile time right?

Reply:
having FirstNamespace is ok imo

Reply:
and nsMode is very rarely set to fallback

Reply:
I tried your suggestion, but it looks like it doesn't work for  the use case mentioned [here](https://github.com/i18next/i18next/pull/1994#issuecomment-1631976127) `TFunction<[string, string]>`:

<img width="789" alt="image" src="https://github.com/i18next/i18next/assets/14237/26480dc1-c4ca-47b0-b231-93f433b0a13f">

Reply:
Is ``TFunction<[string, string]>`` really a use case?
Normaly, you have typed translations so ``TFunction<['realNS', 'realOtherNS']>`` or you don't have typed translations at all

Reply:
I don't know — hope one of the maintainers can answer that question. I was just going based on the error message here: https://github.com/i18next/i18next/pull/1994#issuecomment-1631976127

Reply:
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbwK4GcCmAVKBDAdigG2xmAlwF84AzKCEOAIijWwGMYBaYARgA5c0ADxgMA3AChxVJLnalccGGhQwAFAEpE4uHFZkViRXEoBeOKkw58REmVUBtBtgYAaRgCMGAXXUSdAen8dYLgAPQB+cXIgA

Reply:
I'm one of the maintainers ;-)
Based on the usage I'll argue like this: https://github.com/i18next/i18next/pull/1994#issuecomment-1632955910
So I suggest, we make the change with FirstNamespace... and if that will get problems, we come back to your "never" solution, ok?